tools/astarte_dev_tool/lib/application.ex

```diff
@@ -0,0 +1,32 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule AstarteDevTool.Application do
+  @moduledoc false
+
+  use Application
+
+  @impl true
+  def start(_type, args) do
+    {_opts, _, _} =
+      OptionParser.parse(args, switches: [verbose: :boolean], aliases: [v: :verbose])
+
+    # TODO start application
+    {:ok, self()}
+  end
+end
```

Comment:
We don't need an application.
We can remove:
- `application.ex`
- The reference to the application in `mix.ex`
- The `config` folder.

Actually you can achieve just what is needed by running `mix new astarte_dev_tool` and using the output as the first commit of the PR.

Reply:
@davidebriani the config folder has `logger` configuration..are u sure?

Reply:
`mix new astarte_dev_tool` does not generate a `config` folder, so I assume it's not needed.